clang/lib/Sema/SemaInit.cpp

```diff
@@ -9920,7 +9920,9 @@ QualType Sema::DeduceTemplateSpecializationFromInitializer(
     if (auto *AliasTemplate = dyn_cast_or_null<TypeAliasTemplateDecl>(
             TemplateName.getAsTemplateDecl())) {
       Diag(Kind.getLocation(),
-           diag::warn_cxx17_compat_ctad_for_alias_templates);
+           getLangOpts().CPlusPlus20
```

Comment:
@Sirraide : Didn't we have a function you were making to get the right diagnostic here? Should/could we use that here?

Reply:
Yes we can use `DiagCompat()` here once #132348 is merged (which should happen in a few hours barring unforeseen circumstances). I’ll leave another comment once that has happened.

Reply:
It has been merged since (I had to revert it once but I’ve already relanded it)

Reply:
Ah :)  I was just blaming my bad memory.  I just got back from a week off, so my brain isn't particularly caught up yet :D

Reply:
> Ah :) I was just blaming my bad memory. I just got back from a week off, so my brain isn't particularly caught up yet :D

I merged it a few hours ago so you’re good ;Þ